test/AlgebraicGeometry/ToricVarieties/toric_schemes.jl

```diff
@@ -114,9 +114,42 @@ end
 end
 
 @testset "toric divisors to weil divisors" begin
-  IP = weighted_projective_space(NormalToricVariety, [3, 4, 23])
+  IP = weighted_projective_space(NormalToricVariety, [3, 2, 5])
   w = canonical_divisor(IP)
-  D = Oscar.Oscar.underlying_divisor(w; check=true)
+  D = Oscar.underlying_divisor(w; check=true)
+  D2 = Oscar.underlying_divisor(w; algorithm=:via_polymake, check=true)
+  D3 = Oscar.underlying_divisor(w; algorithm=:via_oscar, check=true)
+  @test D == D2 == D3
+
   @test w == forget_toric_structure(w)
-  @test w + D == 2*w
+  @test w + D == 2*Oscar.underlying_divisor(w)
+
+  prim = Oscar._torusinvariant_weil_divisors(IP; check=true)
+  # Delete the cache manually
+  delete!(IP.__attrs, :_torusinvariant_weil_divisors)
+  prim2 = Oscar._torusinvariant_weil_divisors(IP; check=true, algorithm=:via_oscar)
+  @test prim == prim2
+  @test prim[1] !== prim2[1]
+
+  D = WeilDivisor(Oscar._ideal_sheaf_via_polymake(IP, ZZ.([2, 3, 7])))
+  D2 = 2 * prim[1] + 3 * prim[2] + 7 * prim[3]
+  
+  @test is_effective(D)
+  @test is_effective(D2)
+
+  #@test D == D2 # Test takes too long
+  IP = projective_space(NormalToricVariety, 1)
+  w = canonical_divisor(IP)
+  K0 = Oscar.underlying_divisor(w, algorithm=:via_polymake)
+  delete!(w.__attrs, :underlying_divisor)
+  K1 = Oscar.underlying_divisor(w, algorithm=:direct)
+  delete!(w.__attrs, :underlying_divisor)
+  K2 = Oscar.underlying_divisor(w, algorithm=:via_oscar)
+  delete!(w.__attrs, :underlying_divisor)
+  
+  @test K1 == K2
+  @test K1 == K0
+  @test K2 == K0
```

Comment:
Is this last line necessary? In theory, K1 == K2 and K0 == K1 implies K0 == K2. Maybe this is also true in `OSCAR` (for deeper reasons)?

Reply:
These trigger different calls for the comparison. So they test different code, indeed.